modules/nextflow/src/main/groovy/nextflow/processor/PublishDir.groovy

```diff
@@ -346,6 +347,8 @@ class PublishDir {
             FileHelper.deletePath(destination)
             processFileImpl(source, destination)
         }
+
+        notifyFilePublish(destination)
```

Comment:
Was thinking if it may be better to run this into a separate thread pool

Reply:
I think that there is no need to create a new thread for this. This is only triggering the notification to all the observers, but the writing of the file is already done in another thread (using an Agent). Plus, most of the times (if this is not a symbolic link) we invoke this from a thread that is only running the `safeProcessFile` method.

http://github.com/nextflow-io/nextflow/blob/pipeline-reports/modules/nextflow/src/main/groovy/nextflow/processor/PublishDir.groovy#L289-L295